NorthstarDLL/squirrel/squirrel.cpp

```diff
@@ -200,13 +202,53 @@ template <ScriptContext context> void SquirrelManager<context>::VMCreated(CSquir
 
 			if (dependency.Name == pair.second)
 			{
+				std::string v = dependency.Version;
+				std::vector<SQInteger> semver;
+				size_t last = 0;
+				size_t next = 0;
+
+				size_t rc = v.find('-');
+				if (rc != std::string::npos)
+				{
+					v = v.substr(0, rc);
+				}
```

Comment:
If we want to be truly semver-compliant, the prerelease logic is a lot more complicated, both for splitting, and for comparison.

https://cs.opensource.google/go/x/mod/+/refs/tags/v0.9.0:semver/semver.go
https://semver.org/spec/v2.0.0.html#spec-item-9

Reply:
I can include something like https://github.com/zmarko/semver to replace my very simple parsing logic

Reply:
I think that's going a bit overkill. I'm personally fine with just limiting it to `MAJOR.MINOR.PATCH` and only extending from there if need be ^^

Reply:
Ehh, I'm okay with either (as long as we support whatever thunderstore supports, at a minimum), but we can't call it semver if it isn't.

Reply:
Thunderstore currently only supports `MAJOR.MINOR.PATCH`

https://github.com/thunderstore-io/Thunderstore/issues/634

Reply:
We should support whatever the `version` in the mod.json supports, we shouldn't limit ourselves to just what thunderstore supports, as a thunderstore package != a northstar mod

Reply:
However, they say that "`in any case, we've not ruled out extended semver support`"...

To ensure it doesn't break in the future, we could check the format to ensure it only contains `MAJOR.MINOR.PATCH[-rc#]`, and always count it as a non-match if it doesn't.

Reply:
> However, they say that "`in any case, we've not ruled out extended semver support`"...

This PR should have nothing to do with thunderstore in any capacity, therefore making this irrelevant.

Reply:
I might be misremembering things but iirc the `Version` in the mod.json has nothing to do with the `version` in the thunderstore manifest.

Reply:
> I might be misremembering things but iirc the `Version` in the mod.json has nothing to do with the `version` in the thunderstore manifest.

correct

Reply:
> Ehh, I'm okay with either (as long as we support whatever thunderstore supports, at a minimum), but we can't call it semver if it isn't.

Well let's call it "Dependency Constants that support version comparisons" or whatever, doesn't really matter

Reply:
Ah, that's good to know. I'm not terribly familiar with the modding side of NS, and didn't realize that we version them independently from thunderstore.